appcelerator/titanium_mobile
# fix(ios): amend Ti.UI.Shortcut implementation

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-26818

**Summary:**
- Modified `Ti.UI.Shortcut` to be a module.
- Removed `Ti.UI.createShortcut()` API.
- Temporarily disabled Android shortcut unit tests until it's been changed to match.

**Test:**
1. Build and run the below on iOS.
2. Tap on "Add Shortcut" button.
3. Tap on "Shortcut Exists?" button.
4. Verify alert shows shortcut info.
5. Tap on "List Added Shortcuts" button.
6. Verify an alert lists 1 shortcut.
7. Close the alert dialog.
8. Tap on the home button.
9. Tap and hold the app icon.
10. Verify popup shows "Shortcut Title" option.
11. Tap on the "Shortcut Title" menu item.
12. Verify app is shown in foreground with alert stating shortcut was clicked on.
13. Tap on "Remove 1 Shortcut" button.
14. Tap on "Shortcut Exists?" button.
15. Verify alert states shortcut was **NOT** found.
16. Tap on "List Added Shortcuts" button.
17. Verify alert does not list any shortcuts.
18. Tap on Home button.
19. Tap and hold the app icon.
20. Verify "Shortcut Title" no longer appears in popup.

```javascript
Ti.UI.Shortcut.addEventListener("click", function(e) {
	alert(
		"Clicked on shortcut:\n" +
		"- id: " + e.item.id + "\n" +
		"- title: " + e.item.title + "\n" +
		"- description: " + e.item.description);
});

var shortcutItem = Ti.UI.createShortcutItem({
	id: "shortcut_item_1",
	title: "Shortcut Title",
	description: "Shortcut Description",
	icon: Ti.UI.iOS ? Ti.UI.iOS.SHORTCUT_ICON_TYPE_ADD : undefined,
});

var window = Ti.UI.createWindow({ layout: "vertical" });
var addButton = Ti.UI.createButton({
	title: "Add Shortcut",
	top: Ti.App.iOS ? "30dp" : "10dp",
});
addButton.addEventListener("click", function() {
	Ti.UI.Shortcut.add(shortcutItem);
});
window.add(addButton);
var existsButton = Ti.UI.createButton({
	title: "Shortcut Exist?",
	top: "10dp",
});
existsButton.addEventListener("click", function() {
	var item = Ti.UI.Shortcut.getById(shortcutItem.id);
	if (item) {
		alert(
			"Shortcut '" + item.id + "' was found.\n" +
			"- Title: " + item.title + "\n" +
			"- Description: " + item.description);
	} else {
		alert("Shortcut '" + shortcutItem.id + "' NOT found.");
	}
});
window.add(existsButton);
var removeByReferenceButton = Ti.UI.createButton({
	title: "Remove 1 Shortcut",
	top: "10dp",
});
removeByReferenceButton.addEventListener("click", function() {
	Ti.UI.Shortcut.remove(shortcutItem);
});
window.add(removeByReferenceButton);
var removeAllButton = Ti.UI.createButton({
	title: "Remove All Shortcuts",
	top: "10dp",
});
removeAllButton.addEventListener("click", function() {
	Ti.UI.Shortcut.removeAll();
});
window.add(removeAllButton);
var getAllButton = Ti.UI.createButton({
	title: "List Added Shortcuts",
	top: "10dp",
});
getAllButton.addEventListener("click", function() {
	let message = "Ti.UI.Shortcut.items:\n";
	for (const nextItem of Ti.UI.Shortcut.items) {
		message += "- " + nextItem.id + "\n";
	}
	alert(message);
});
window.add(getAllButton);
window.open();
```
